Add location to runs

daveshah opened the pull request.

daveshah commented on web/static/vendor/maps.js:
```diff
@@ -0,0 +1,80 @@
+
+function initMap() {
+    var map = new google.maps.Map(document.getElementById('map'), {
+        center: {lat: -33.8688, lng: 151.2195},
```
This should be set by the users browser if possible.

daveshah commented on web/static/vendor/maps.js:
```diff
@@ -0,0 +1,80 @@
+            ].join(' ');
+        }
+
+        infowindow.setContent('<div><strong>' + place.id + '</strong><br>' + address);
```
We'll want to store off ```place.id``` for form submission and display mo' better info on the map instead.

daveshah commented on web/static/vendor/maps.js:
```diff
@@ -0,0 +1,80 @@
+    setupClickListener('changetype-address', ['address']);
+    setupClickListener('changetype-establishment', ['establishment']);
+    setupClickListener('changetype-geocode', ['geocode']);
+
```
definitely don't need all of these for now.

daveshah commented on web/views/run_view.ex:
```diff
@@ -1,3 +1,8 @@
   use Gatherto.Web, :view
+
+  def google_maps_api do
+    api_key = System.get_env("GOOGLE_MAPS_API_KEY")
```
I don't know how I feel about this being the way it is in the RunView module. (Doesn't feel right)

matt-keibler commented on web/views/run_view.ex:
```diff
@@ -23,4 +22,8 @@ defmodule Gatherto.RunView do
   end
 
+  def google_maps_api do
+    api_key = System.get_env("GOOGLE_MAPS_API_KEY")
```
Do we need to update the documentation to tell devs how to generate a maps API key or is it the same as the other key?

daveshah replied:
We'll most definitely want to update the docs. Good callout!

daveshah replied:
@matt-keibler I updated the docs. I hope they make sense :)

daveshah commented on priv/repo/migrations/20160906195240_add_location_id_to_runs.exs:
```diff
@@ -0,0 +1,9 @@
+      add :location_id, :string
+    end
+  end
+end
```
I opted to keep it simple and just add a location_id to runs since right now the run model would just be 2 ids - a local id and a Google place id. If this changes in the future we can always migrate these to a new model.

daveshah commented on web/static/css/rnnr.css:
```diff
@@ -19,3 +19,54 @@
+    font-family: Roboto;
+    font-size: 13px;
+    font-weight: 300;
+}
```
This could probably be cleaned up a bit but I thought we could leave styling for a later task.

daveshah commented on web/static/vendor/maps.js:
```diff
@@ -0,0 +1,101 @@
+function initMap() {
```
Ugh - so, I spent some time cleaning this up but don't want to go too crazy with this. Curious to know thoughts on it as it stands.

matt-keibler replied:
If we aren't planning on customizing the functionality too much, I think it is fine.

daveshah committed:
Add location to runs

Runs now have locations provided by the Google Places API. A runs
location_id is currently a Google place_id for simplicity.

Closes #29

daveshah commented:
I squashed this down to 1 commit. Please let me know if this looks good enough to merge in or if there's anything that needs to change / needs clarity / anything I didn't think of. Thanks!

matt-keibler commented:
Looks good to me 👍
